stemgen/__main__.py

```diff
@@ -230,12 +244,12 @@ def generate(
             continue
         click.echo(f"Processing {filename}...")
 
-        src = Track(file)
+        src = Track(file, demucs.sample_rate)
         samples = src.read()
         original, stems = None, []
 
         with click.progressbar(
-            length=samples.shape[1] * len(demucs.weights),
+            length=demucs.length(samples) * shifts,
```

Comment:
I believe I had made that fix but it must have been lost across some other local branch - thanks for bringing this back!